tests/test_flagsmith.py

```diff
@@ -191,6 +226,46 @@ def test_get_identity_flags_uses_local_environment_when_available(
     assert identity_flags[0].value == "some-feature-state-value"
 
 
+def test_get_identity_flags_includes_segments_in_evaluation_context(
+    mocker: MockerFixture,
+    flagsmith: Flagsmith,
+    evaluation_context: SDKEvaluationContext,
+) -> None:
+    # Given
+    flagsmith._evaluation_context = evaluation_context
+    flagsmith.enable_local_evaluation = True
+    mock_engine = mocker.patch("flagsmith.flagsmith.engine")
```

Comment:
Mock specific function and use `autospec`:


```suggestion
    mock_get_evaluation_result = mocker.patch(
        "flagsmith.flagsmith.engine.get_evaluation_result",
        autospec=True,
    )
```